grafana/deploy.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+
+# This is a super simple approach to syncing dashboards
+# TODO:
+#  - check if dashboard really needs update (will cause many versions in grafana db)
+#  - check if stale folder needs to be deleted
+#  - check if stale dashboard needs to be deleted 
+
+set -euo pipefail
+
+RESOURCEGROUP=${GLOBAL_RESOURCEGROUP:-global}
+DRY_RUN=${DRY_RUN:-false}
+
+if [[ -z ${GRAFANA_NAME} ]];
+then
+    echo "GRAFANA_NAME needs to be set"
+    exit 1
+fi
+
+cd dashboards
+
+existing_folders=$(az grafana folder list -g ${RESOURCEGROUP}  -n ${GRAFANA_NAME})
+
+for d in $(find . -mindepth 1 -maxdepth 1 -type d -exec basename {} \;)
+do
+    if [[ $(echo $existing_folders | jq --arg TITLE "$d"  -c '.[] | select( .title == $TITLE )' |wc -l )  -gt 0 ]];
+    then
+        folderUid=$(echo $existing_folders | jq --arg TITLE "$d"  -r '.[] | select( .title == $TITLE ) | .uid' )
+    else
+        if [[ ${DRY_RUN} == "true" ]];
+        then
+            echo "would create folder '${d}' on managed grafana ${GRAFANA_NAME} in rg ${RESOURCEGROUP}"
+        else
+            folderUid=$(az grafana folder create --only-show-errors  -g ${RESOURCEGROUP}  -n ${GRAFANA_NAME} --title ${d} |jq -r '.uid')
+        fi
+    fi
+    pushd ${d}
+    IFS=$'\n'; for dashboard in $(ls -1)
```

Comment:
same as the other suggestion but for files
```suggestion
   for file in $(find . -mindepth 1 -maxdepth 1 -type f)
```

I had not seen this inner loop earlier, but now that I have I think you could simplify the two loops into one

```sh
cd dashboards
find . -mindepth 1 -maxdepth 2 -type f | while read -r file; do
  dashboard_dir=$(basename "$(dirname "$file")")
  dashboard_file=$(basename "$file")
  # handle creating/updating dashboards as before...
done
```

not deal-breaking by any means, but switching to a `find ... | while read` will also make it properly handle when/if there are spaces in the directory or file names which I had not thought about in my earlier suggestions

Reply:
As discussed in chat lets leave it as is for now